spec/classes/init_spec.rb

```diff
@@ -339,6 +341,29 @@
       end
     end
 
+    netgroup_validations = {
+      :valid   => %w(NSS NIS OFF),
+      :invalid => [ true, 1, 'undef', 'nss', 'nis', 'off' ],
+    }
+
+    netgroup_validations[:valid].each do |valid|
+      context "with vas_conf_vasd_netgroup_mode set to #{valid}" do
+        let(:params) { { :vas_conf_vasd_netgroup_mode => valid, } }
+        it do
+          should contain_file('vas_config').with_content(/netgroup-mode = #{valid}/)
+        end
+      end
+    end
+
+    netgroup_validations[:invalid].each do |invalid|
+      context "with vas_conf_vasd_netgroup_mode set to #{invalid}" do
+        let(:params) { { :vas_conf_vasd_netgroup_mode => invalid, } }
+        it 'should fail' do
+          expect { should contain_class(subject) }.to raise_error(Puppet::Error, /Valid values are NSS, NIS and OFF|is not a string/)
+        end
+      end
+    end
+
```

Comment:
Please use the 'variable type and content validations' section at the bottom for the type and content validations.

With that in place, atomic tests of your functionality are good enough. Take the tests for $join_domain_controllers as example.